rotkehlchen/utils/datadir.py

```diff
@@ -0,0 +1,136 @@
+    for path in paths:
+        target_path_dir = target_dir / path.name
+        try:
+            move(path, target_path_dir)
+        except (PermissionError, Error) as e:
+            log.error(f'During restructuring data directory failed to copy {path} to {target_path_dir} due to {e}')  # noqa: E501
+
+
+def _handle_images(data_dir: Path, icons_path: Path | None) -> None:
+    if icons_path is not None:
+        # create the images data directory and move paths there
+        img_data_dir = _create_directory_with_potential_backup(data_dir, IMAGESDIR_NAME)
+        assets_dir = img_data_dir / ASSETIMAGESDIR_NAME
+        assets_dir.mkdir(parents=True, exist_ok=True)
+
+        targetpath = assets_dir / ALLASSETIMAGESDIR_NAME
+        try:
+            move(icons_path, targetpath)
+        except (PermissionError, Error) as e:
+            log.error(f'During restructuring data directory failed to move {icons_path} to {targetpath} due to {e}')  # noqa: E501
+            return
+
+        all_assets_dir = targetpath
+        for name, target_dir in ((AVATARIMAGESDIR_NAME, img_data_dir), (CUSTOMASSETIMAGESDIR_NAME, assets_dir)):  # noqa: E501
+            subdir = all_assets_dir / name
+            if subdir.exists():
+                targetpath = target_dir / name
+                try:
+                    move(subdir, targetpath)
+                except (PermissionError, Error) as e:
+                    log.error(f'During restructuring data directory failed to move {subdir} to {targetpath} due to {e}')  # noqa: E501
+                    continue
+
+
+def maybe_restructure_rotki_data_directory(data_dir: Path) -> None:
+    """Restructures the data directory to contain users under a specficic subdirectory
+    and other kind of data under other ones. Essentially implement:
+    https://github.com/rotki/rotki/issues/4841
+
+    TODO: This should stay here for only a few versions and later be removed. This
+    restructuring can always be done by the user in a super weird edge case, by hand.
+    """
+    if (data_dir / USERSDIR_NAME).is_dir() and (data_dir / GLOBALDIR_NAME).is_dir():
+        return  # already restructured
+
+    user_paths = []
+    other_paths = []
+    global_path = None
+    airdrops_path = None
+    airdrops_poap_path = None
+    misc_path = None
+    icons_path = None
+    for x in data_dir.iterdir():
+        try:
+            if x.is_dir() and (x / USERDB_NAME).exists():
+                user_paths.append(x)
+            elif x.is_dir() and x.name == 'global_data':
+                global_path = x
+            elif x.is_dir() and x.name == 'airdrops':
+                airdrops_path = x
+            elif x.is_dir() and x.name == 'airdrops_poap':
+                airdrops_poap_path = x
+            elif x.is_dir() and x.name == MISCDIR_NAME:
+                misc_path = x
+            elif x.is_dir() and x.name == 'icons':
+                icons_path = x
+            else:
+                other_paths.append(x)
```

Comment:
isn't it better to do

```suggestion
            if not x.is_dir():
                other_paths.append(x)
                continue
  
            if  (x / USERDB_NAME).exists():
                user_paths.append(x)
            elif x.name == 'global_data':
                global_path = x
            elif x.name == 'airdrops':
                airdrops_path = x
            elif x.name == 'airdrops_poap':
                airdrops_poap_path = x
            elif x.name == MISCDIR_NAME:
                misc_path = x
            elif x.name == 'icons':
                icons_path = x
            else:
                other_paths.append(x)
```

Reply:
no it's not. The point is to move anything not fiting into other paths. Including directories.

Reply:
Okay I will add it. It's trivial difference. A bit harder to read, but also perhaps a tiny bit faster due to the x.is_dir() check. Had not seen the else when I made the comment above.